api/access/server.go

```diff
@@ -161,12 +162,73 @@ func (s *Server) SendTransaction(ctx context.Context, in *access.SendTransaction
 	return nil, errors.New("not implemented")
 }
 
-func (s *Server) GetTransaction(ctx context.Context, in *access.GetTransactionRequest) (*access.TransactionResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetTransaction(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResponse, error) {
+	txID := flow.HashToID(in.Id)
+	tb, err := s.index.Transaction(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction: %w", err)
+	}
+
+	resp := access.TransactionResponse{
+		Transaction: convert.TransactionToMessage(*tb),
+	}
+
+	return &resp, nil
 }
 
-func (s *Server) GetTransactionResult(ctx context.Context, in *access.GetTransactionRequest) (*access.TransactionResultResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetTransactionResult(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResultResponse, error) {
+	txID := flow.HashToID(in.Id)
+	tx, err := s.index.Transaction(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction: %w", err)
+	}
+
+	result, err := s.index.Result(txID)
+	if err != nil && !errors.Is(err, storage.ErrNotFound) {
+		return nil, fmt.Errorf("could not retrieve transaction result: %w", err)
+	}
+
+	// We also need the height of the transaction we're looking at.
+	height, err := s.index.HeightForTransaction(tx.ID())
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block height: %w", err)
+	}
+
+	block, err := s.index.Header(height)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block header: %w", err)
+	}
+	blockID := block.ID()
+
+	statusCode := uint32(0)
+	if result != nil && result.ErrorMessage == "" {
```

Comment:
Saying it once more, but I don't think we should have any `nil` checks in our code base. If there is a `nil` result, we should just consider the transaction not to not exist at the moment. And even if you want to do that check, I would use an `errors.Is` sentinel error check, rather than returning a `nil` value from `Result()`.